vllm/model_executor/models/llama.py

```diff
@@ -102,8 +102,9 @@ def __init__(
         assert self.total_num_heads % tp_size == 0
         self.num_heads = self.total_num_heads // tp_size
         self.total_num_kv_heads = num_kv_heads
-        assert self.total_num_kv_heads % tp_size == 0
-        self.num_kv_heads = self.total_num_kv_heads // tp_size
+        assert max(self.total_num_kv_heads, tp_size)% min(self.total_num_kv_heads, tp_size) == 0
```

Comment:
Could you comment on the logic here?

Reply:
A little bit longer but easier to understand:
```suggestion
        if self.total_num_kv_heads >= tp_size:
            assert self.total_num_kv_heads % tp_size == 0
        else:
            assert tp_size % self.total_num_kv_heads == 0
```